groonga: add msgpack-c as default dependency

eth3lbert opened the pull request:
pgroonga requires groonga with msgpack #272202.
Should close #272202.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

eth3lbert committed:
pgroonga: build groonga with msgpack-c

pgroonga requires groonga with msgpack #272202.

eth3lbert commented:
Result of `nixpkgs-review pr 272203` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>postgresql12JitPackages.pgroonga</li>
    <li>postgresql12Packages.pgroonga</li>
    <li>postgresql13JitPackages.pgroonga</li>
    <li>postgresql13Packages.pgroonga</li>
    <li>postgresql14JitPackages.pgroonga</li>
    <li>postgresql14Packages.pgroonga</li>
    <li>postgresqlJitPackages.pgroonga (postgresql15JitPackages.pgroonga)</li>
    <li>postgresql15Packages.pgroonga</li>
    <li>postgresql16JitPackages.pgroonga</li>
    <li>postgresql16Packages.pgroonga</li>
  </ul>
</details>

NickCao commented on pkgs/servers/sql/postgresql/ext/pgroonga.nix:
```diff
@@ -10,7 +10,17 @@ stdenv.mkDerivation rec {
-  buildInputs = [ postgresql msgpack-c groonga ];
+  buildInputs =
+    let
+      groonga-with-msgpack = groonga.overrideAttrs (oa: {
```
Or should it be `groonga.override { suggestSupport = true; }`?

eth3lbert replied:
`pgroonga` doesn't seem to use the suggest feature from `groonga`, would you recommend that we still use `suggestSupport` flag to build `groonga`?

NickCao replied:
According to the upstream documentation: https://groonga.org/docs/install/debian.html#build-from-source, msgpack should be added as a dependency by default. So another choice would be to include `msgpack` (or `msgpack-c` if supported) in `buildInputs` unconditionally.

eth3lbert replied:
I second this, It also a default dependency according to groonga's [brew formula](https://formulae.brew.sh/formula/groonga).

eth3lbert replied:
Have added msgpack-c as default dependency, thanks for the review.

eth3lbert committed:
Revert "pgroonga: build groonga with msgpack-c"

This reverts commit 234b6136566eb37342fd02af9493041b7bfa01d6.
Based on the discussion and upstream documentation, it seems more
reasonable to add msgpack as a default dependency for groonga.

eth3lbert committed:
groonga: add msgpack-c as default dependency

According to the upstream documentation: https://groonga.org/docs/install/debian.html#build-from-source,
msgpack should be added as a dependency by default.

eth3lbert commented:
Result of `nixpkgs-review` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>groonga</li>
    <li>postgresql12JitPackages.pgroonga</li>
    <li>postgresql12Packages.pgroonga</li>
    <li>postgresql13JitPackages.pgroonga</li>
    <li>postgresql13Packages.pgroonga</li>
    <li>postgresql14JitPackages.pgroonga</li>
    <li>postgresql14Packages.pgroonga</li>
    <li>postgresqlJitPackages.pgroonga (postgresql15JitPackages.pgroonga)</li>
    <li>postgresql15Packages.pgroonga</li>
    <li>postgresql16JitPackages.pgroonga</li>
    <li>postgresql16Packages.pgroonga</li>
  </ul>
</details>

NickCao reviewed (approved):
diff LGTM, checked build log, `msgpack-c` is picked up.

Result of `nixpkgs-review pr 272203` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>groonga</li>
    <li>postgresql12JitPackages.pgroonga</li>
    <li>postgresql12Packages.pgroonga</li>
    <li>postgresql13JitPackages.pgroonga</li>
    <li>postgresql13Packages.pgroonga</li>
    <li>postgresql14JitPackages.pgroonga</li>
    <li>postgresql14Packages.pgroonga</li>
    <li>postgresqlJitPackages.pgroonga (postgresql15JitPackages.pgroonga)</li>
    <li>postgresqlPackages.pgroonga (postgresql15Packages.pgroonga)</li>
    <li>postgresql16JitPackages.pgroonga</li>
    <li>postgresql16Packages.pgroonga</li>
  </ul>
</details>

cc @ericsagnes
